### ISSUE TITLE
Scorpion Weaponry

### ISSUE THREAD

**Issue by HellDragon-HK** [GreatEmerald/UT3Vehicles#128]
- SOUND RELATED
  - [ ] Blades when broke still have the extend and retract sound in UT3 but not in UT2004
  - [ ] UT3ScorpionBallBlue.uc
 ImpactSound=Sound'UT3Weapons2.BioRifle.BioRifleExplode' // GEm: FIXME, should not depend on UT3Weapons2
  - [X] Extra sound effect noticed when firing turret that is not heard in UT3

 - VISUAL EFFECTS RELATED
  - [ ] Turret fire has a very short green muzzle flash in UT2004, in UT3 it's a bright large red flash with electricity
  - [ ] Turret projectile is just a ball of energy in UT2004 and not a solid black ball with energy inside it like in UT3.....possibly add modified Bio Rifle projectile visual?
  - [ ] Projectile in UT3 has a rocket like explosion

**Comment by GreatEmerald** [comment]
Broken blades not making a sound is a feature as far as I'm concerned.

Is the ball explosion sound actually from the biorifle in UT3?

Yes, the visuals need to be ported over for the ball. I just experimented a bit with it to make it look nicer than the EONS one, but didn't do actual porting of it.

**Comment by HellDragon-HK** [comment]
Yeah sounds like the same thing in both games to me, maybe a little louder in UT3 though and looks like it says it in the code too

defaultproperties
{
	MyDamageType=class'UTDmgType_ScorpionGlob'


	ProjFlightTemplate=ParticleSystem'VH_Scorpion.Effects.P_Scorpion_Bounce_Projectile'
	ProjExplosionTemplate=ParticleSystem'VH_Scorpion.Effects.PS_Scorpion_Gun_Impact'


	Explosionsound=SoundCue'A_Weapon_BioRifle.Weapon.A_BioRifle_FireImpactExplode_Cue'
	ImpactSound=SoundCue'A_Weapon_BioRifle.Weapon.A_BioRifle_FireImpactFizzle_Cue'
}

I didn't think you could port the visuals and thought they needed to be done by hand like you said U2 particles needed to be, if they are easy to port awesome, looking forward to it

**Comment by GreatEmerald** [comment]
Hah, OK. Looks like Epic got lazy with that projectile :D

Well, I think the ball is a mesh, not a particle emitter. IIRC what I did with the projectile is that I simply applied a ball mesh from UT2004, Bombing Run. What's missing is the correct texture for it.

**Comment by HellDragon-HK** [comment]
Maybe they just ran out of time, either way I still believe the biorifle sound was a decent choice, what would you have made it sound like?

I noticed it's the bombing run one, I've seen a lot of mods do something similar.

**Comment by GreatEmerald** [comment]
Probably. I would have made a new sound for it and probably not have made it be a ball to begin with.

**Comment by HellDragon-HK** [comment]
New Issue, the blades are very hard to break off and can go through objects easily....I'm guessing the collision on them didn't update with the rest of the Scorpion when we changed it to UT3's scale size

Also GE have you ever thought maybe the Scorpion's projectile is a variant of bio goo, it has a goo look about it in UT3 I think and the fire sound (not the impact) kind of works for that thought as well....and there's the fact it's called ScorpionGlob as well.

EDIT - Fixed it, will add to request
